src/node/snapchain_node.rs

```diff
@@ -223,4 +229,24 @@ impl SnapchainNode {
             warn!("No actor found for shard, could not forward message");
         }
     }
+
+    pub async fn handle_block(&self, block: Block) {
+        for chunk in block.shard_chunks {
+            let header = chunk.header.expect("Expects chunk to have a header");
+            let height = header.height.expect("Expects header to have a height");
+            let mempool = self
+                .shard_mempools
+                .get(&height.shard_index)
+                .expect("Expects mempool to exist for shard");
+            let mut mempool_write = mempool.write().await;
+            for transaction in chunk.transactions {
+                for user_message in transaction.user_messages {
+                    mempool_write.remove(user_message.hex_hash());
```

Comment:
See note about locks. I would structure this as one single channel for receiving confirmed shard chunks that's then published to a channel that the mempool listens too. In a select loop within the mempool, evict the messages that were included in the block. 

The other thing to consider is how to evict messages that we rejected before being included in the block (e.g. validation failed or rate limits were exceeded etc). We the mempool channel might need to be more generic than just accepting shard chunks.